### DISCUSSION TITLE
Does a rejection for public key needs to be based on its algorithm or its bits?

### DISCUSSION THREAD

**Discussion by martgil** [openpgpjs/openpgpjs#1742] [Q&A] [answered]
Hello @twiss,

Good afternoon OpenPGP.js team!

I am curious about how OpenPGP.js considers weak key types. I am a new learner here, so I apologize if this question is nonsensical or inappropriate.

I know that the team decided to reject ElGamal and DSA key types based on the following links here:

https://github.com/openpgpjs/openpgpjs/blob/026b348cf87988cf95896dd9c5dd1a0c8129ddb1/src/config/config.js#L247 https://github.com/openpgpjs/openpgpjs/blob/026b348cf87988cf95896dd9c5dd1a0c8129ddb1/src/key/helper.js#L397

But I'd like to know when it will be considered a weak key. For example, would it still be correct to consider an ElGamal key weak if it is composed of 4096 bits?

Answers mean a lot to me. thank you in advance!

**Comment by larabr** [comment, marked as answer]
ElGamal and DSA have been deprecated by the crypto-refresh on security grounds (see https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-13.html#section-12.5 and https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-13.html#section-12.6), so it's not based on the key size for them 🙂

**Reply by martgil** [reply]
Very informative! Thank you very much, have a great day!
